central/sensor/service/pipeline/nodeindex/three_pipelines_test.go

```diff
@@ -0,0 +1,597 @@
+	}{
+		"lone node index should not find the node in the DB": {
+			operations: []func(t *testing.T, np pipeline.Fragment, ninvp pipeline.Fragment, nidxp pipeline.Fragment) error{
+				// V4 node-scan (node index) for node1 arrives
+				func(t *testing.T, np pipeline.Fragment, ninvp pipeline.Fragment, nidxp pipeline.Fragment) error {
+					return nidxp.Run(context.Background(), clusterID, createNodeIndexMsg(nodeID, "v3"), nil)
+				},
+			},
+			setUpMocksAndEnv: func(t *testing.T, m *usedMocks) {
+				t.Setenv(features.ScannerV4.EnvVar(), "true")
+				t.Setenv(env.NodeIndexEnabled.EnvVar(), "true")
+				gomock.InOrder(
+					// node index arrives
+					m.nodeDatastore.EXPECT().GetNode(gomock.Any(), gomock.Eq(nodeID)).MinTimes(1).Return(nil, false, nil),
+				)
+			},
+			wantNodeExists: false,
+		},
+
+		"node index arriving after node should result in data from the node being overwritten": {
+			operations: []func(t *testing.T, np pipeline.Fragment, ninvp pipeline.Fragment, nidxp pipeline.Fragment) error{
+				// V1 node-scan for node1 arrives over the node pipeline
+				func(t *testing.T, np pipeline.Fragment, ninvp pipeline.Fragment, nidxp pipeline.Fragment) error {
+					return np.Run(context.Background(), clusterID, createNodeMsg(nodeID, "v1"), nil)
+				},
+				// V4 node-scan (node index) for node1 arrives over the node index pipeline
+				func(t *testing.T, np pipeline.Fragment, ninvp pipeline.Fragment, nidxp pipeline.Fragment) error {
+					return nidxp.Run(context.Background(), clusterID, createNodeIndexMsg(nodeID, "v4"), nil)
+				},
+			},
+			setUpMocksAndEnv: func(t *testing.T, m *usedMocks) {
+				t.Setenv(features.ScannerV4.EnvVar(), "true")
+				t.Setenv(env.NodeIndexEnabled.EnvVar(), "true")
+				gomock.InOrder(
+					// node arrives
+					m.clusterStore.EXPECT().GetClusterName(gomock.Any(), gomock.Eq(clusterID)).Times(1).Return(clusterID, true, nil),
+					m.cveDatastore.EXPECT().EnrichNodeWithSuppressedCVEs(gomock.Any()),
+					m.riskStorage.EXPECT().UpsertRisk(gomock.Any(), gomock.Any()).AnyTimes().Return(nil),
+					m.nodeDatastore.EXPECT().UpsertNode(gomock.Any(), gomock.Any()).AnyTimes().Return(nil),
+					// node index arrives
+					m.nodeDatastore.EXPECT().GetNode(gomock.Any(), gomock.Eq(nodeID)).AnyTimes().Return(nodeWithScore, true, nil),
+					m.cveDatastore.EXPECT().EnrichNodeWithSuppressedCVEs(gomock.Any()).AnyTimes().Return(),
+					m.riskStorage.EXPECT().UpsertRisk(gomock.Any(), gomock.Any()).AnyTimes().Return(nil),
+					m.nodeDatastore.EXPECT().UpsertNode(gomock.Any(), nodeWithScanWithKernelV4).AnyTimes().Return(nil),
+					// check what got stored in the DB
+					m.nodeDatastore.EXPECT().GetNode(gomock.Any(), gomock.Eq(nodeID)).AnyTimes().Return(nodeWithScanWithKernelV4, true, nil),
+				)
+			},
+			wantNodeExists:            true,
+			wantKernelVersionNode:     "v1",
+			wantKernelVersionNodeScan: "v4",
+		},
+
+		"node index arriving after node inventory should result in inventory being lost": {
```

Comment:
Yeah, they are lost, but I think that "overwritten" describes the action more precisely
```suggestion
		"node index arriving after node inventory should result in inventory being overwritten": {
```